R/score_forecasts.R

```diff
@@ -71,27 +74,24 @@ score_forecasts <- function(
   
   # score using scoringutil
   observation_cols <- c(
-    "model", "location",
+    "model", 
+    "location", 
     "horizon", "temporal_resolution", "target_variable",
     "forecast_date", "target_end_date"
   )
-  scores <- tibble::tibble(scoringutils::eval_forecasts(data = joint_df, by = observation_cols))
-  
-  # mangle the data for output
-  # if users didn't provide score types
-  if (!missing(desired_score_types)) {
-    extra_na_colnames <- setdiff(desired_score_types, colnames(scores))
-    extra_na_cols <- matrix(
-      NA_real_,
-      nrow = nrow(scores),
-      ncol = length(extra_na_colnames),
-      dimnames = list(NULL, extra_na_colnames)
-    )
-    
-    scores <- scores %>%
-      dplyr::select(c(observation_cols, desired_score_types)) %>%
-      dplyr::bind_cols(as.data.frame(extra_na_cols))
-  }
+
+  scores <- tibble::tibble(scoringutils::eval_forecasts(data = joint_df, 
+    by = observation_cols,
+    summarise_by = c(observation_cols, "range")),
+    interval_score_arguments = list(weigh = TRUE)) %>%
+    tidyr::pivot_wider(id_cols = observation_cols,
+      names_from = c("range"), 
+      values_from = c("coverage", "interval_score", "aem")) %>%
+    ## before next line: do we need to check to ensure aem_0 is a valid column name
+    dplyr::rename(abs_error = aem_0) %>% 
```

Comment:
haven't tested it, but if existence of aem_0 is of concern, we could maybe also do something like 
`dplyr::mutate(abs_error = unique(dplyr::starts_with("aem_"))` 
(not sure that exactly works.) But aem should be the same number for all quantiles.